rfcs/0067-quarterly-releases.md

```diff
@@ -0,0 +1,67 @@
+
+* Three weeks prior to the release
+    * The release manager will cut a release branch and create a new tag
+    * The release manager will audit the licenses of all project dependencies to ensure they are compatible with Apache
+    * The release manager will open a GitHub issue announcing the release branch cut and target date for the release and state that any further inclusions in the release must be manually cherry-picked
+    * The release manager will create a PR targeted to merge into the release branch with the necessary changes to make a release (i.e. changing version numbers)
+    * Contributors that wish changes to be cherry-picked should comment on the announcement issue with the relevant PRs and commits and their reasoning. The release manager has final say on which changes should be included but should aim to be inclusive at this stage
+    * The release manager begins gathering release notes
+
+* Two weeks prior to release
+    * Cherry picks become limited to critical changes only
+    * The release manager begins building binaries and testing them against TVM's test suite
+    * The release manager sends the release notes and plan to a vote thread for approval (feedback thread will be open for one week)
+    * The release manager performs another license audit of all project dependencies to ensure they are compatible with Apache
+
+* One week prior to release
+    * The release manager sends the release and binaries to a vote thread for approval (thread open for one week). If rejected the building and test process must be started anew and repeated until the PMC approves.
+
+* Day of the release
+    * The release manager publishes the relevant binaries
+    * The release manager closes the release issue
+    * The release manager makes a GitHub release and updates an in-repo file `RELEASE.md` on both `main` and the release branch with the release notes
+
+Much of this can be automated via GitHub Actions on the apache/tvm repo. Eventually (though maybe not for the upcoming release) these will handle all the building, testing, and publishing of releases so the job of the release manager will become simpler over time.
+
+The release manager will use the release notes and discussions with developers to determine the next version number. Releases will continue the current versioning scheme of `major.minor.patch`, with a typical release involving a bump of the minor release version. Patch versions will be used for follow up releases onto a quarterly release, but not for the next quarter's release. The release manager will be responsible for maintaining the release until the next release, which mainly entails putting up a patch release if necessary. The release manager may also delegate this responsibility to another party if both agree.
```

Comment:
This looks like semver, and will likely be interpreted as semver, but is not. I would strongly encourage considering officially adopting and adhering to semver to help manage API expectations for groups trying to use TVM in production. Time-based releases don't always fit perfectly with semantic versioning, but careful consideration of API changes and additions should make adherence possible. https://semver.org

Reply:
Another alternative is adopting a date-based version number such as `year.quarter.minor`, which will meet the needs for resolving dependencies but does not imply the strict API compatibility requirements across releases that semver has.

Reply:
i agree we are not semver here. however, we are also not 1.0, so it might be worth reconsidering our versioning scheme once we do hit 1.0. I'd like to see some more infrastructure and processes developed (e.g. feature flags and a more codified set of expectations for large projects like metascheduler which are currently developed in-tree and merged in an effort-specific way) before we can commit to semver.

we could consider e.g. `0.202206.0`, though it's a bit hard to parse. i don't think we should propose to rev the major version past 1.0 at this time, though.

Reply:
agree semver is probably too much for us right now and we should revisit at 1.0. Increasing version numbers seem simplest to think about which is what this RFC proposes and anecdotally date releases aren't as common for Python packages (though they are listed in [PEP 440](https://peps.python.org/pep-0440/))